diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -102,7 +105,17 @@ class TrailingCommaRule(configRules: List<RulesConfig>) : DiktatRule(
             // we should write type of node in warning, to make it easier for user to find the parameter
             TRAILING_COMMA.warnAndFix(configRules, emitWarn, isFixMode, "after ${this.elementType}: ${this.text}", this.startOffset, this) {
                 val parent = this.treeParent
-                parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)
+
+                // In case, when we got VALUE_PARAMETER, it may contain comments, which follows the actual parameter and all of them are actually in the same node
+                // Ex: `class A(val a: Int, val b: Int  // comment)`
+                // `val b: Int  // comment` --> the whole expression is VALUE_PARAMETER
+                // So, in this case we must insert comma before the comment, in other cases we will insert it after current node
+                val comments = listOf(EOL_COMMENT, BLOCK_COMMENT, KDOC)
+                val firstCommentNodeOrNull = if (this.elementType == VALUE_PARAMETER) this.children().firstOrNull { it.elementType in comments } else null
+                firstCommentNodeOrNull?.let {
+                    this.addChild(LeafPsiElement(COMMA, ","), firstCommentNodeOrNull)
+                }
+                    ?: parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)
```

Comment:
```suggestion
           if (this.elementType == VALUE_PARAMETER) {
                // In case, when we got VALUE_PARAMETER, it may contain comments, which follows the actual parameter and all of them are actually in the same node
                // Ex: `class A(val a: Int, val b: Int  // comment)`
                // `val b: Int  // comment` --> the whole expression is VALUE_PARAMETER
                // So, in this case we must insert comma before the comment, in other cases we will insert it after current node
                val comments = listOf(EOL_COMMENT, BLOCK_COMMENT, KDOC)
                val firstCommentNodeOrNull = this.children().firstOrNull { it.elementType in comments }
                this.addChild(LeafPsiElement(COMMA, ","), firstCommentNodeOrNull ?: treeNext)
            } else {
                parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)
            }
```
Why not simply like this? Even though it requires additional `?: treeNext` in the first branch, this way we completely separate logic with comments from nodes with other types

Reply:
But in case of `VALUE_PARAMETER` **with** comments we need to add child to `this`: `this.addChild(...)`, but when **there is no** comments or the type **is not** `VALUE_PARAMETER` we need to add child to `parent`: `parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)`.

So your variant is not really suitable

Reply:
Oh, ok, didn't notice.